Fix rate-limiting

MuriloDalRi opened the pull request:
Dependapanda has stopped notifying Slack since the end of March.

It seems like Github is being more aggressive in rate-limiting requests to their API.

This PR attempts to fix this by making API requests more spaced.

[Trello](https://trello.com/c/NpeMpcep/2886-fix-dependapanda-again-5)

ChrisBAshton commented on lib/gateways/pull_request.rb:
```diff
@@ -14,7 +14,7 @@ def execute
 
     def approved_pull_requests
-      query = "is:pr user:alphagov state:open author:app/dependabot author:app/dependabot-preview review:approved"
+      query = "is:pr user:alphagov state:open author:app/dependabot review:approved"
```
Nice improvement 👍 
Nitpick: could you link to some evidence in the commit message?

MuriloDalRi replied:
I added a link to a PR that has a bit more info

ChrisBAshton commented on Rakefile:
```diff
@@ -45,19 +45,32 @@ end
-  puts "fetched repos"
-  sleep 10
+  puts "Finished fetching repos."
+  puts "Sleeping for 60 seconds to avoid rate-limiting"
+
+  sleep 60
```
Might be worth moving this out to a `def avoid_rate_limit` as we use it a few times.

MuriloDalRi replied:
I tried that but Rubocop wasn't happy

ChrisBAshton commented on spec/acceptance/app_spec.rb:
```diff
@@ -220,7 +220,7 @@ def app
   context "Stats page" do
     before do
-      stub_request(:get, "https://api.github.com/search/issues?q=is:pr%20user:alphagov%20author:app/dependabot")
+      stub_request(:get, "https://api.github.com/search/issues?q=is:pr%20user:alphagov%20state:open%20author:app/dependabot%20archived:false")
```
Can you explain (in the commit message) why these changes are necessary? Does the Octokit SDK now append these extra parameters to requests?

ChrisBAshton commented on README.md:
```diff
@@ -38,7 +38,6 @@ Running this will start your application at [localhost:9292](localhost:9292)
 ### Environment variables
 
 - `GITHUB_TOKEN` - OAuth token generated on GitHub which does not require any special permissions
-  - Used to interact with the GitHub API, although not required it will help avoid limiting
```
Good call 👍

ChrisBAshton commented on lib/gateways/pull_request.rb:
```diff
@@ -14,7 +14,7 @@ def execute
 
     def approved_pull_requests
-      query = "is:pr user:alphagov state:open author:app/dependabot review:approved"
+      query = "is:pr user:alphagov state:open author:app/dependabot archived:false review:approved"
```
Ah - I'm guessing the "Fix tests" commit is because of these changes. Can you squash them together?

MuriloDalRi replied:
Done

ChrisBAshton reviewed (commented):
Looking good 👍  has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?

MuriloDalRi commented:
> Looking good 👍 has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?



> Looking good 👍 has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?

I've manually pushed this to Heroku and also tested it locally and it seems to be working. I think I've addressed all comments now if you want to have another look

ChrisBAshton reviewed (approved):
LGTM 👍 great job!
